docs/docs/09.3-environments.md

```diff
@@ -0,0 +1,81 @@
+---
+id: environments
+title: Server-side Environments
+permalink: environments.html
+prev: package-management.html
+next: addons.html
+---
+
+One of the great things about React is that it doesn't require the DOM as a dependency, which means it is possible to render a React application on the server and send the HTML markup down to the client. There are a few things that React expects, so this guide will help you get started in your preferred environment.
+
+
+## Nashorn
+
+Nashorn is a lightweight high-performance JavaScript runtime that runs within the JVM. React should run out of the box in Java8+.
+
+Example:
+
+```java
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import javax.script.ScriptEngine;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+
+import org.apache.commons.io.IOUtils;  // https://commons.apache.org/proper/commons-io/download_io.cgi
+
+public class ReactRender
+{
+  public static void main(String[] args) throws ScriptException, IOException {
+    ScriptEngine nashorn = new ScriptEngineManager().getEngineByName("nashorn");
+
+    nashorn.eval("var global = this;"); // react expects global to be defined
+
+    nashorn.eval(getScript("https://fb.me/react-15.0.1.js"));
+    nashorn.eval(getScript("https://fb.me/react-dom-server-15.0.1.js"));
+
+    System.out.println(nashorn.eval("ReactDOMServer.renderToString(React.createElement('div', {}, 'Hello World!'));"));
+  }
+
+  public static String getScript(String url) throws MalformedURLException, IOException {
+    InputStream stream = new URL(url).openStream();
+    try {
+      return IOUtils.toString(stream);
+    }
+    finally {
+      stream.close();
+    }
+  }
+}
+```
+
+If your application uses node modules, or want to transform JSX in Nashorn, you will need to do some additional environment setup. The following resources may be helpful in getting you started:
+
+ * [http://winterbe.com/posts/2015/02/16/isomorphic-react-webapps-on-the-jvm/](http://winterbe.com/posts/2015/02/16/isomorphic-react-webapps-on-the-jvm/)
+ * [https://github.com/nodyn/jvm-npm](https://github.com/nodyn/jvm-npm)
+ * [https://gist.github.com/aesteve/883e0fd33390451cb8eb](https://gist.github.com/aesteve/883e0fd33390451cb8eb)
+
+ > NOTE: Using Babel within Nashorn will require Java 8u72+, as update 72 fixed [JDK-8135190](https://bugs.openjdk.java.net/browse/JDK-8135190).
+
+
+
+## Node
```

Comment:
Let's not put words in my mouth. This isn't really a list in the traditional sense. This is a page full of information and Node is approximately 100000000x more important that Nashorn. Please reorder these.